rpcs3/Emu/Memory/vm.cpp

```diff
@@ -664,6 +684,15 @@ namespace vm
 			// Check ref counter (using unused member info for it)
 			if (shm->info == 2)
 			{
+				// Allocate shm object for itself
+				u64 shm_self = reinterpret_cast<u64>(shm->map_self()) ^ range_locked;
+
+				// Pre-set range-locked flag (real pointers are 47 bits)
+				// 1. To simplify range_lock logic
+				// 2. To make sure it never overlaps with 32-bit addresses
+				// Also check that it's aligned (lowest 16 bits)
+				verify(HERE), (shm_self & 0xffff'8000'0000'ffff) == range_locked;
```

Comment:
Cant it be sign extended or something? im not sure if operating systems allow such "negative" pointers in userspace buts its not guaranteed by x86.

Reply:
It seems that negative address space of the same size is reserved for kernel memory, so we will never see it in userspace.